### PR TITLE
[MERGE] add new module product_lifecycle that manage the data model of the replacement of discontinued products

### PR DESCRIPTION
**Pull request by zaoral**
Task [vx#3453](https://www.vauxoo.com/#id=3453&view_type=form&model=project.task)
Task [vx#3649](https://www.vauxoo.com/#id=3649&view_type=form&model=project.task)

 HU [314](https://www.vauxoo.com/?debug=#id=314&view_type=form&model=user.story&action=726) Sprint 12

Runbot: Manual Pre-Build  http://runbot.vauxoo.com/runbot/team/yoytec-24#102

### CODE REVIEW

**Comment by zaoral** [comment]
waiting for testing tools results

**Comment by zaoral** [comment]
@nhomar please review this PR

travis is red, but for external reasons to this PR

#### Note: This PR must be merge after Vauxoo/addons-vauxoo#291 is merge

**Comment by zaoral** [comment]
I new issue odoo/odoo#6994 was made to discuss change of the product.template state field to the product.product model.

File: product_lifecycle/__openerp__.py [hunk 1]
```diff
@@ -0,0 +1,52 @@
...
+    "name": "Product Lifecycle",
+    "summary": "Manage replacement of discontinued products",
+    "version": "1.0",
+    "author": "Vauxoo",
+    "website": "http://www.vauxoo.com/",
+    "category": "product",
+    "depends": [
+        "product",
+        "purchase",
+        "purchase_order_line_sequence",
```
Comment (nhomar) [inline product_lifecycle/__openerp__.py:36]: No necesitas las dos dependencias de arriba. purchase_order_line sequence depende de purchase y a su vez este de product.

Por otra parte no entiendo por que purchase_order_line_sequence es necesaria para éste feature, me comentas por que?

Saludos.

**Comment by nhomar** [comment]
@zaoral Please:
1. Add unit test following all the product lifecycle to ensure always this module is tested mixed with all others which manipulate products.
2. Can you make a video with this feature working please?

Regards.

File: product_lifecycle/__openerp__.py (hunk 1 above)
Comment (zaoral) [inline product_lifecycle/__openerp__.py:36]: @nhomar 
- I will remove the unnecessary dependencies.
- This functionality we use it on the product_lifecycle module that need for the identificate the purchase order lines to replace and to indicate to the user what lines have discontinued products.

**Comment by zaoral** [comment]
@nhomar ok. working on the unit test and the video. Please approve this guideline for the video https://drive.google.com/a/vauxoo.com/file/d/0B1UUUA6bRx8-MzRCc2ZBN054Tkk/view

**Comment by zaoral** [comment]
A merge with the new unit test was made to this branch.

Waiting for runbot result in http://runbot.vauxoo.com/runbot/team/yoytec-24#48pull_266
At this moment travis result is red it have importing errors external to this PR

**Comment by zaoral** [comment]
Waiting for combined runbot result http://runbot.vauxoo.com/runbot/team/yoytec-24#102

**Comment by zaoral** [comment]
@nhomar PR ready. Runbot green http://runbot.vauxoo.com/runbot/team/yoytec-24#102 for this PR and for https://github.com/Vauxoo/yoytec/pull/124. Please review this two PRs. First this one and next the https://github.com/Vauxoo/yoytec/pull/124

**Pull request merged**